### PR TITLE
Regenerate and checkin corpus files

### PR DESCRIPTION
**Pull request by Alan-Jowett**
Signed-off-by: Alan Jowett <alanjo@microsoft.com>

Resolves: #1848 
Resolves: #1505

## Description

Regenerate the corpus for the execution context fuzzer to add support for the prog run method.

## Testing

CI/CD

## Documentation

No.

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -13,3 +13,6 @@
 *.vcxproj           text eol=crlf
 *.vcxproj.filters   text eol=crlf
 *.o                 -text
+
+# Mark generated corpus files as binary.
+**/corpus/**       -text
```
Comment (dthaler) [inline .gitattributes:18]: nit: looks like `-text` doesn't line up with line 15

File: tests/end_to_end/test_helper.cpp [hunk 2]
```diff
@@ -147,6 +152,46 @@ class duplicate_handles_table_t
...
+    }
+    return value;
+}
+
+/**
+ * @brief Get an environment variable as a boolean.
+ *
+ * @param[in] name Environment variable name.
+ * @return false Environment variable is set to "false", "0", or if it's not set.
+ * @return true Environment variable is set to any other value.
```
Comment (dthaler) [inline tests/end_to_end/test_helper.cpp:174]:
```suggestion
 * @retval false Environment variable is set to "false", "0", or if it's not set.
 * @retval true Environment variable is set to any other value.
```

File: docs/Fuzzing.md [hunk 3]
```diff
@@ -47,4 +47,18 @@ An alternative of running an admin CMD is to reproduce a crash to use the local
 ```
 verifier_fuzzer.exe <crash-file-name>
 ```
-This method will show the line of crash in the source file.
\ No newline at end of file
+This method will show the line of crash in the source file.
+
+# Regenerating the corpus for execution context fuzzer
+The libfuzzer tests work best if they have an existing
+[corpus](https://releases.llvm.org/15.0.0/docs/LibFuzzer.html#corpus) to work
```
Comment (dthaler) [inline docs/Fuzzing.md:54]:
```suggestion
[corpus](https://releases.llvm.org/11.0.0/docs/LibFuzzer.html#corpus) to work
```
we use v11 currently not v15

File: netebpfext/net_ebpf_ext_sock_addr.c [hunk 4]
```diff
@@ -451,8 +451,8 @@ _net_ebpf_ext_connection_context_initialize_value(
  * @param[in] addr1 Pointer to first sock_addr struct to compare.
  * @param[in] addr2 Pointer to second sock_addr struct to compare.
 
- * @return true if the addresses are same.
-   @return false otherwise.
+ * @retval true if the addresses are same.
+   @retval false if the addresses are different.
```
Comment (dthaler) [inline netebpfext/net_ebpf_ext_sock_addr.c:455]:
```suggestion
 * @retval true The addresses are same.
   @retval false The addresses are different.
```

File: tests/libs/util/hash.h [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+#include <string>
+#include <vector>
+
+typedef class _hash
+{
+  public:
+    typedef std::vector<std::tuple<const uint8_t*, size_t>> byte_range_t;
+    template <typename T>
+    static void
+    append_byte_range(byte_range_t& byte_range, const T& value)
```
[the remark this answers is not in the document]
Comment (Alan-Jowett) [reply tests/libs/util/hash.h:12]: These aren't pointer values so I don't think code analysis annotations will do anything.

[the remark this answers is not in the document]
Comment (dthaler) [reply tests/libs/util/hash.h:12]: Yes they are, lvalue references are still references.  https://en.cppreference.com/w/cpp/language/reference

File: libs/platform/ebpf_bitmap.h [hunk 6]
```diff
@@ -45,7 +46,8 @@ extern "C"
      * @param[in, out] bitmap Pointer to the bitmap.
      * @param[in] index Index to modify.
      * @param[in] interlocked Perform the operation using interlocked.
-     * @return true if the bit was set, false otherwise.
+     * @retval true if the bit was modified.
+     * @retval false if the bit was already set.
```
Comment (dv-msft) [inline libs/platform/ebpf_bitmap.h:50]: This might read better as "false if the bit was already _clear_/_reset_"

Comment (dv-msft) [inline libs/platform/ebpf_bitmap.h:50]: This might read better as "false if the bit was already reset"

**Review by dthaler — APPROVED** [review]

**Review by dv-msft — APPROVED** [review]

**Pull request merged**